pkg/skaffold/sync/sync_test.go

```diff
@@ -921,58 +921,64 @@ var pod = &v1.Pod{
 }
 
 func TestPerform(t *testing.T) {
-	tests := []struct {
-		description string
-		image       string
-		files       syncMap
-		pod         *v1.Pod
-		cmdFn       func(context.Context, v1.Pod, v1.Container, syncMap) *exec.Cmd
-		cmdErr      error
-		clientErr   error
-		expected    []string
-		shouldErr   bool
+	tests := map[string]struct {
+		image     string
+		files     syncMap
+		pod       *v1.Pod
+		cmdFn     func(context.Context, v1.Pod, v1.Container, syncMap) *exec.Cmd
+		cmdErr    error
+		clientErr error
+		expected  []string
+		shouldErr bool
 	}{
-		{
-			description: "no error",
-			image:       "gcr.io/k8s-skaffold:123",
-			files:       syncMap{"test.go": {"/test.go"}},
-			pod:         pod,
-			cmdFn:       fakeCmd,
-			expected:    []string{"copy test.go /test.go"},
+		"no error": {
+			image:    "gcr.io/k8s-skaffold:123",
+			files:    syncMap{"test.go": {"/test.go"}},
+			pod:      runningPod,
+			cmdFn:    fakeCmd,
+			expected: []string{"copy test.go /test.go"},
 		},
-		{
-			description: "cmd error",
-			image:       "gcr.io/k8s-skaffold:123",
-			files:       syncMap{"test.go": {"/test.go"}},
-			pod:         pod,
-			cmdFn:       fakeCmd,
-			cmdErr:      fmt.Errorf(""),
-			shouldErr:   true,
+		"cmd error": {
+			image:     "gcr.io/k8s-skaffold:123",
+			files:     syncMap{"test.go": {"/test.go"}},
+			pod:       runningPod,
+			cmdFn:     fakeCmd,
+			cmdErr:    fmt.Errorf(""),
+			shouldErr: true,
 		},
-		{
-			description: "client error",
-			image:       "gcr.io/k8s-skaffold:123",
-			files:       syncMap{"test.go": {"/test.go"}},
-			pod:         pod,
-			cmdFn:       fakeCmd,
-			clientErr:   fmt.Errorf(""),
-			shouldErr:   true,
+		"client error": {
+			image:     "gcr.io/k8s-skaffold:123",
+			files:     syncMap{"test.go": {"/test.go"}},
+			pod:       runningPod,
+			cmdFn:     fakeCmd,
+			clientErr: fmt.Errorf(""),
+			shouldErr: true,
 		},
-		{
-			description: "no copy",
-			image:       "gcr.io/different-pod:123",
-			files:       syncMap{"test.go": {"/test.go"}},
-			pod:         pod,
-			cmdFn:       fakeCmd,
-			shouldErr:   true,
+		"pod not running": {
+			image:     "gcr.io/k8s-skaffold:123",
+			files:     syncMap{"test.go": {"/test.go"}},
+			pod:       nil,
```

Comment:
in the code you're only logging a warning, but here the test should fail?

Reply:
it fails because no files were synced